akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -568,14 +568,18 @@ private[akka] class Shard(
   }
 
   def onEntitiesRemembered(ids: Set[EntityId]): Unit = {
-    log.debug("{}: Shard initialized", typeName)
     if (ids.nonEmpty) {
       entities.alreadyRemembered(ids)
       log.debug("{}: Restarting set of [{}] entities", typeName, ids.size)
       context.actorOf(
         RememberEntityStarter.props(context.parent, self, shardId, ids, settings),
         "RememberEntitiesStarter")
     }
+    shardInitialized()
+  }
+
+  private def shardInitialized(): Unit = {
+    log.debug("{}: Shard initialized", typeName)
```

Comment:
The only difference here is that we now first log about the remembered entities and then we log "Shard initialized". I don't think it's an issue though. The remembered entities restarting is an async process, so it's not something that is ready when we move to `idle` behavior.